ksql-api/src/main/java/io/confluent/ksql/api/server/ConnectionQueryManager.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpConnection;
+import io.vertx.core.http.HttpServerRequest;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class ConnectionQueryManager {
+
+  private final Context context;
+  private final Server server;
+  private final Map<HttpConnection, ConnectionQueries> connectionsMap = new HashMap<>();
+
+  public ConnectionQueryManager(final Context context, final Server server) {
+    this.context = Objects.requireNonNull(context);
+    this.server = Objects.requireNonNull(server);
+  }
+
+  public PushQueryHolder createApiQuery(final QuerySubscriber querySubscriber,
+      final HttpServerRequest request) {
+    checkContext();
+    final ConnectionQueries connectionQueries = getConnectionQueries(request);
+    final PushQueryHolder query = new PushQueryHolder(server, querySubscriber,
+        connectionQueries::removeQuery);
+    connectionQueries.addQuery(query);
+    return query;
+  }
+
+  private ConnectionQueries getConnectionQueries(final HttpServerRequest request) {
+    final HttpConnection conn = request.connection();
+    ConnectionQueries connectionQueries = connectionsMap.get(conn);
+    if (connectionQueries == null) {
+      connectionQueries = new ConnectionQueries(conn);
+      connectionsMap.put(conn, connectionQueries);
+    }
```

Comment:
FYI, with reference to https://github.com/confluentinc/ksql/pull/4376#discussion_r370584768, using `computeIfAbsent` does only a single lookup into the map, where as this code does two: one in `get` and one in `put`.  


Bit surprised you prefer the more verbose / old-school approach, rather than making use of the newer features of the language, given you're the one pushing for using a more modern JDK version :p

Reply:
It's just that if I was to use computeIfAbsent I would probably spend at least 30 seconds looking up the API to figure out what to pass it, whereas with this I can knock it out from memory in about 5 seconds. Yes, I'm old school ;)

Reply:
Spending 30 seconds now will pay dividends in the future.